File: INE5416/haskell/atividade5/q12.hs
-- Crie uma fun ̧c ̃ao com assinaturaapagar ::  Int -> [t] -> [t], a qual recebe um n ́umero de elementos,
-- uma lista,  e  retorna uma lista.  Esta  fun ̧c ̃ao  deve remover da lista os  n primeiros elementos 
-- fornecidoscomo parˆametro.  Por exemplo, a chamada(apagar 3 [1,2,3,4,5])deve retornar[4,5].
-- N ̃ao utilizenenhuma fun ̧c ̃ao pronta to Haskell para esta tarefa

apagar :: Int -> [t] -> [t]
apagar 0 a = a
apagar n (a:b) = apagar (n-1) b

main = do
    let l = [1..10]
    print (apagar 5 l)

File: INE5416/prolog/atividade16/q12.pl
apagar(0, L1, L1) :- !.
apagar(N, [_|T], L2) :- Nm1 is N-1, apagar(Nm1, T, L2).

% ?- apagar(0, [1, 2 , 3], L).
% L = [1, 2, 3].

% ?- apagar(1, [1, 2 , 3], L).
% L = [2, 3].

% ?- apagar(2, [1, 2 , 3], L).
% L = [3].

% ?- apagar(3, [1, 2 , 3], L).
% L = [].